db/channel_cache.go

```diff
@@ -107,7 +106,7 @@ func newChannelCache(dbName string, backgroundTasks []BackgroundTask, terminator
 	if err != nil {
 		return nil, err
 	}
-	backgroundTasks = append(backgroundTasks, bgt)
+	*backgroundTasks = append(*backgroundTasks, bgt)
```

Comment:
It seems like `channelCache` should have its own blocking `Stop` method for this one task. It makes it a bit easier to reason about if there's no mutable lists of tasks being passed around between different caches.

This new stop method can be called from `changeCache.Stop()`, as well as for standalone usages of `channelCache`, like all of the `TestChannelCache*` tests, which currently can't stop this background task either.

Reply:
This change was included in the latest commit. I had considered this approach by reusing the waitForBGTCompletion method attached on database context, but later decided to not go with it because I've given precedence to keep the change minimal. When looking at the suggestion from you and Adam, I see this will be more cleaner.